bookkeeper-server/src/main/java/org/apache/bookkeeper/replication/AuditorTask.java

```diff
@@ -0,0 +1,176 @@
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.bookkeeper.replication;
+
+import com.google.common.collect.Lists;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.LongAdder;
+import org.apache.bookkeeper.client.BKException;
+import org.apache.bookkeeper.client.BookKeeper;
+import org.apache.bookkeeper.client.BookKeeperAdmin;
+import org.apache.bookkeeper.common.concurrent.FutureUtils;
+import org.apache.bookkeeper.conf.ClientConfiguration;
+import org.apache.bookkeeper.conf.ServerConfiguration;
+import org.apache.bookkeeper.meta.LedgerManager;
+import org.apache.bookkeeper.meta.LedgerUnderreplicationManager;
+import org.apache.bookkeeper.net.BookieId;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+abstract class AuditorTask implements Runnable {
+    private static final Logger LOG = LoggerFactory.getLogger(AuditorTask.class);
+
+    protected final ServerConfiguration conf;
+    protected AuditorStats auditorStats;
+    protected BookKeeperAdmin admin;
+    protected LedgerManager ledgerManager;
+    protected LedgerUnderreplicationManager ledgerUnderreplicationManager;
+    private final SubmitTaskHandler submitTaskHandler;
+    private final ShutdownTaskHandler shutdownTaskHandler;
+
+    AuditorTask(ServerConfiguration conf,
+                AuditorStats auditorStats,
+                BookKeeperAdmin admin,
+                LedgerManager ledgerManager,
+                LedgerUnderreplicationManager ledgerUnderreplicationManager,
+                SubmitTaskHandler submitTaskHandler,
+                ShutdownTaskHandler shutdownTaskHandler) {
+        this.conf = conf;
+        this.auditorStats = auditorStats;
+        this.admin = admin;
+        this.ledgerManager = ledgerManager;
+        this.ledgerUnderreplicationManager = ledgerUnderreplicationManager;
+        this.submitTaskHandler = submitTaskHandler;
+        this.shutdownTaskHandler = shutdownTaskHandler;
+    }
+
+    @Override
+    public void run() {
+        runTask();
+    }
+
+    protected abstract void runTask();
+
+    protected void waitIfLedgerReplicationDisabled() throws ReplicationException.UnavailableException,
```

Comment:
It only invokes by AuditorBookieCheckTask, move it to AuditorBookieCheckTask.

Reply:
Addressed done.